pkg/cluster/util.go

```diff
@@ -398,8 +411,8 @@ func (c *Cluster) labelsSelector() *metav1.LabelSelector {
 	return &metav1.LabelSelector{MatchLabels: c.labelsSet(false), MatchExpressions: nil}
 }
 
-func (c *Cluster) roleLabelsSet(role PostgresRole) labels.Set {
-	lbls := c.labelsSet(false)
+func (c *Cluster) roleLabelsSet(shouldAddExtraLabels bool, role PostgresRole) labels.Set {
```

Comment:
this function name starts to become questionable.

should we rethink this? inheritLabels(roleLalbeSet(role)) or a later merge of labels?

Reply:
I'll let you resolve this discussion if you agree that this should be done in a separated PR.